DEVELOPING.md

```diff
@@ -0,0 +1,36 @@
+# Developer's Guide
+
+We use Visual Studio Code for developing LoopBack and recommend the same to our
+users.
+
+## VSCode setup
+
+Install the following extensions:
+
+ - [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+ - [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+
+## Development workflow
+
+### Visual Studio Code
+
+1. Start the build task (Cmd+Shift+B) to run TypeScript compiler in the
+   background, watching and recompiling files as you change them. Compilation
+   errors will be shown in the VSCode's "PROBLEMS" window.
```

Comment:
:robot: OpenAI

Please add a brief description of what the TypeScript compiler does.
```suggestion
# Developer's Guide

We use Visual Studio Code for developing LoopBack and recommend the same to our users.

## VSCode setup

Install the following extensions:

- [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint): provides linting capabilities for JavaScript and TypeScript code.
- [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode): provides code formatting capabilities.

## Development workflow

### Visual Studio Code

1. Start the build task (Cmd+Shift+B) to run TypeScript compiler in the background, watching and recompiling files as you change them. The TypeScript compiler compiles TypeScript code into JavaScript code. Compilation errors will be shown in the VSCode's "PROBLEMS" window.

2. Execute "Run Rest Task" from the Command Palette (Cmd+Shift+P) to re-run the test suite and lint the code for both programming and style errors. Linting errors will be shown in VSCode's "PROBLEMS" window. Failed tests are printed to terminal output only.

### Other editors/IDEs

1. Open a new terminal window/tab and start the continuous build process via `npm run build:watch`. It will run TypeScript compiler in watch mode, recompiling files as you change them. Any compilation errors will be printed to the terminal.

2

<!-- This is an auto-generated comment by OpenAI -->